adobe/spectrum-web-components
# Help text

## Description
Add `<sp-help-text>` pattern.
- ship `HelpTextManager` helper class to support delivery of Help Text in other elements
- update `<sp-field-group>`, `<sp-number-field>`, `<sp-radio-group>`, `<sp-search>`, and `<sp-textfield>` to support this pattern
- add `findDescribedNode` helper to the unit testing suite to cover over eccentricities in WebKit support for described content

## Related issue(s)

- fixes #1704

## Motivation and context
If it's in Spectrum _AND_ Spectrum CSS it should likely be here! Also, we want to make your forms even nicer looking.

## How has this been tested?
    1. Go to:
      - https://help-text--spectrum-web-components.netlify.app/components/help-text/
      - https://help-text--spectrum-web-components.netlify.app/components/field-group/
      - https://help-text--spectrum-web-components.netlify.app/components/radio/
      - https://help-text--spectrum-web-components.netlify.app/components/textfield/
      - https://help-text--spectrum-web-components.netlify.app/components/textarea/
    2. Experience the new element and its alignment with https://opensource.adobe.com/spectrum-css/helptext.html and https://spectrum.adobe.com/page/help-text/
    1. Go to https://help-text--spectrum-web-components.netlify.app/components/help-text/
    2. With the screen reader, make sure that the Help Text is read in association with the form element that it is placed within AS WELL AS singularly
    3. Toggle the validity to ensure that this content live updates as well

## Types of changes

## Checklist

CC: @davidkudera
